drv/fpga-api/src/lib.rs

```diff
@@ -293,14 +307,51 @@ pub fn await_fpga_ready(
 /// Load a bitstream.
 pub fn load_bitstream(
     fpga: &mut Fpga,
-    data: &[u8],
+    auxflash: &mut AuxFlash,
+    blob: AuxFlashBlob,
     bitstream_type: BitstreamType,
-    chunk_len: usize,
+    expected_checksum: [u8; 32],
 ) -> Result<(), FpgaError> {
-    let mut bitstream = fpga.start_bitstream_load(bitstream_type)?;
+    // Do an initial read to verify the checksum
+    let mut scratch_buf = [0u8; 128];
+    let mut pos = blob.start;
+    let mut sha = Sha3_256::new();
+    while pos != blob.end {
```

Comment:
Tiny nit - maybe `<` instead of `!=`? Two reasons:

1. Slightly clearer that this loop is increasing `pos` until it gets to `blob.end`
2. If we screwed up the increment of `pos` and shoot past `blob.end`, we'll break out as soon as that happens

Reply:
👍🏻  I ended up using your abstraction from below, which uses `<`